org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/CompletionHandler.java

```diff
@@ -39,17 +39,38 @@
 import org.eclipse.lsp4j.CompletionParams;
 import org.eclipse.lsp4j.jsonrpc.messages.Either;
 
+import com.google.common.collect.Sets;
+
 public class CompletionHandler{
 
 	public final static CompletionOptions DEFAULT_COMPLETION_OPTIONS = new CompletionOptions(Boolean.TRUE, Arrays.asList(".", "@", "#", "*"));
 	private static final Set<String> UNSUPPORTED_RESOURCES = Sets.newHashSet("module-info.java", "package-info.java");
 
+	private static final Comparator<CompletionItem> PROPOSAL_COMPARATOR = new Comparator<CompletionItem>() {
+
+		@Override
+		public int compare(CompletionItem o1, CompletionItem o2) {
+			return getSortText(o1).compareTo(getSortText(o2));
+		}
+
+		private String getSortText(CompletionItem ci) {
+			return StringUtils.defaultString(ci.getSortText(), "99999999999");
```

Comment:
Can we use the `SortTextHelper.CELING` for consistency?
Its value is `999_999_999`.

Reply:
done